🚀Preview Release

CalvinWilkinson opened the pull request:
<!--
    !! NOTE !! - ONLY PROJECT OWNERS AND MAINTAINERS CAN CREATE PRODUCTION AND PREVIEW RELEASE PULL REQUESTS
    Please use the "preview-release-pr" pull request template if you have contributions to make.
-->
<!--suppress HtmlDeprecatedAttribute -->
<h1 style="font-weight:bold" align="center">Preview Release Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

- [x] The **_base(target)_** branch for this pull request is a **_release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: release/v<major>.<minor>.<patch>
      Example: release/v1.2.3
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_release_** branch with the correct naming syntax in the following example.
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: release/v<major>.<minor>.<patch>
    Example: release/v1.2.3
    ```
  </details>

💡For more information on branching, refer to the project's [branching documentation](../../Documentation/Branching.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title is set to _**🚀Preview Release**_.

- [x] A `🚀Preview Release` label has been added to this pull request. 

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] A QA issue has been created and manual QA testing has been performed for the changes being released _(if applicable)_.

CalvinWilkinson committed:
Start of issue #147

CalvinWilkinson committed:
Update branching diagram

CalvinWilkinson committed:
Start work for issue #283

CalvinWilkinson committed:
Implement PR #288 Changes

CalvinWilkinson committed:
Add and update badges in README file

Commit:
feat: fix #282

CalvinWilkinson committed:
Implement PR #288 Changes

KinsonDigitalAdmin committed:
Merge pull request #288 from KinsonDigital/preview/feature/147-update-branching-docs

Update Branching Documentation

CalvinWilkinson committed:
Remove content left behind

CalvinWilkinson committed:
Add additional person to README

CalvinWilkinson committed:
Remove horizontal rules

CalvinWilkinson committed:
Reorganize badges and make small improvements

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/283-edit-readme-badges

CalvinWilkinson committed:
Implement PR #289 Changes

kselena committed:
Implement PR #289 Changes

ibergx00 committed:
Update .github/ISSUE_TEMPLATE/bug-issue-template.yml

Co-authored-by: Calvin Wilkinson (Admin) <105235056+KinsonDigitalAdmin@users.noreply.github.com>

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/282-fix-spelling-error-in-issue-template

KinsonDigitalAdmin committed:
Merge pull request #291 from Kshitiz-Mhto/preview/feature/282-fix-spelling-error-in-issue-template

Fixed spelling error in issue template

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/283-edit-readme-badges

CalvinWilkinson committed:
Merge pull request #289 from KinsonDigital/preview/feature/283-edit-readme-badges

Edit badges in README

CalvinWilkinson committed:
Start work for issue #280

CalvinWilkinson committed:
Update runsettings file for Visual Studio users

CalvinWilkinson committed:
Update preview version

CalvinWilkinson committed:
Add the letter 'v' before the version in the release notes for v1.0.0-preview.9

Commit:
feat: fix #295

CalvinWilkinson committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/295-fix-issue-template-spelling-issue

Commit:
feat: fixes #296

KinsonDigitalAdmin committed:
Merge pull request #299 from Kshitiz-Mhto/preview/feature/295-fix-issue-template-spelling-issue

Fix issue template spelling issue

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/296-fix-project-item-issue-template

CalvinWilkinson committed:
Merge pull request #300 from Kshitiz-Mhto/preview/feature/296-fix-project-item-issue-template

Fixed project item issue template 
Great job @Kshitiz-Mhto!!

Thank you for your contribution!!

Commit:
feat: closes #240

CalvinWilkinson committed:
Implement PR #297 Changes

CalvinWilkinson committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/280-update-runsettings-file

CalvinWilkinson committed:
Disable preview and prod release branch status checks

CalvinWilkinson committed:
Merge pull request #297 from KinsonDigital/preview/feature/280-update-runsettings-file

Update runsettings file

CalvinWilkinson committed:
Start work for issue #188

kselena committed:
start work for issue #302

CalvinWilkinson committed:
Refactor and improve existing release notes

CalvinWilkinson committed:
Add preview and production release note templates

kselena committed:
Refactor and adjust branching docs

CalvinWilkinson committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/240-create-codeowners-file

CalvinWilkinson committed:
Merge pull request #305 from Kshitiz-Mhto/preview/feature/240-create-codeowners-file

Added CODEOWNERS file to the project

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/188-improve-release-notes

CalvinWilkinson committed:
Implement PR #303 Changes

CalvinWilkinson committed:
Disable branch status check workflows

CalvinWilkinson committed:
Start work for issue #286

CalvinWilkinson committed:
Adjust team shared settings for code style with private and static field naming

CalvinWilkinson committed:
Stylecop field naming rule added to solution level editor config

CalvinWilkinson committed:
Make adjustments to solution level editorconfig file

CalvinWilkinson committed:
Merge pull request #306 from KinsonDigital/preview/feature/286-adjust-code-style-setup

Adjust code naming style setup

CalvinWilkinson committed:
Fix couple more issues with release not templates

CalvinWilkinson committed:
Update previous preview release notes

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/188-improve-release-notes

kselena committed:
Implement PR #303 Changes

CalvinWilkinson committed:
Merge pull request #303 from KinsonDigital/preview/feature/188-improve-release-notes

Improve release notes

kselena committed:
Start work for issue #312

kselena committed:
Update feature request issue template

kselena committed:
Update project item issue template

kselena committed:
Update qa testing template

kselena committed:
Update bug issue template

kselena committed:
Update release todo issue template

kselena committed:
Update research issue template

kselena committed:
Update feature pr template

kselena committed:
Update preview feature pr template

CalvinWilkinson committed:
Implement PR #313 Changes

CalvinWilkinson committed:
Implement PR #313 Changes

CalvinWilkinson committed:
Merge pull request #313 from KinsonDigital/preview/feature/312-update-issue-template

Update issue and pr template

Commit:
feat: update readme file #310

kselena committed:
Start work for issue #316

kselena committed:
Rename PR templates

kselena committed:
Rearrange TODO items

KinsonDigitalAdmin committed:
Merge pull request #317 from KinsonDigital/preview/feature/316-rename-pr-templates

Rename PR template files

KinsonDigitalAdmin committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/302-improve-branch-docs

CalvinWilkinson committed:
Merge pull request #304 from KinsonDigital/preview/feature/302-improve-branch-docs

Improve branching documentation

kselena committed:
Start work for issue #309

kselena committed:
Update code of conduct

KinsonDigitalAdmin committed:
Merge pull request #319 from KinsonDigital/preview/feature/309-update-code-conduct

Update Code of Conduct

CalvinWilkinson committed:
Start work for issue #89

CalvinWilkinson committed:
Add project logo as an icon file to the project

CalvinWilkinson committed:
Add the logo icon to the csproj for the nuget package to use

CalvinWilkinson committed:
Add additional changes to pack icon into nuget package

KinsonDigitalAdmin committed:
Merge pull request #321 from KinsonDigital/preview/feature/89-create-logo-icon

Create logo icon to use for nuget.org

ibergx00 committed:
Update README.md

i mistakenly removed the ` . `

kselena committed:
Start work for issue #325

kselena committed:
Update release todo template by removing backslashes

CalvinWilkinson committed:
Merge pull request #326 from KinsonDigital/preview/feature/325-fix-release-todo-template

Fix release todo issue template

kselena committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/310-update-readme-file

KinsonDigitalAdmin committed:
Merge pull request #324 from Kshitiz-Mhto/preview/feature/310-update-readme-file

Update readme file

kselena committed:
Start work for issue 327

kselena committed:
Update release todo template

CalvinWilkinson committed:
Add extra space to normalize double spacing across file

KinsonDigitalAdmin committed:
Merge pull request #331 from KinsonDigital/preview/feature/327-update-release-todo-template

Update release todo template

kselena committed:
Start work for issue 333

kselena committed:
Add/rearrange tasks in PR templates

kselena committed:
Start work for issue #331

kselena committed:
Removed issue and pr templates, config, funding and codeowners

CalvinWilkinson committed:
Add 2 tasks to PR templates (#334)

kselena committed:
Implement PR #336 Changes

kselena committed:
temp

kselena committed:
Merge branch 'preview/v1.0.0-preview.10' into preview/feature/335-remove-issue-pr-templates

CalvinWilkinson committed:
Remove issue and pr templates (#336)

CalvinWilkinson committed:
Start for work issue #293

kselena committed:
Start work for issue 338

kselena committed:
Remove qa testing issue template and folder

kselena committed:
Start work for issue 339

kselena committed:
Update triage workflow

CalvinWilkinson committed:
Remove qa testing issue template (#340)

kselena committed:
Implement PR #339 Changes

CalvinWilkinson committed:
Merge pull request #341 from KinsonDigital/preview/feature/339-update-triage-workflow

Update triage issue workflow

CalvinWilkinson committed:
📦Update nuget package coverlet.msbuild from version v3.1.0 to v3.1.2

CalvinWilkinson committed:
📦Update nuget package Microsoft.NET.Test.Sdk from version v16.10.0 to v16.11.0

CalvinWilkinson committed:
📦Update nuget package Moq from version v4.16.1 to v4.18.2

CalvinWilkinson committed:
📦Update nuget package NVorbis from version v0.10.3 to v0.10.4

CalvinWilkinson committed:
📦Removed nuget package NVorbis

This is not required anymore.  Sound is now managed in CASL.

CalvinWilkinson committed:
📦Update nuget package xunit from version v2.4.1 to v2.4.2

CalvinWilkinson committed:
📦Update nuget package xunit.runner.visualstudio from version v2.4.3 to v2.4.5

CalvinWilkinson committed:
📦Update nuget package SimpleInjector from version v5.3.2 to v5.4.0

CalvinWilkinson committed:
📦Update nuget package Microsoft.NET.Test.Sdk from version v16.11.0 to v17.3.0

CalvinWilkinson committed:
📦Update nuget package System.IO.Abstractions from version v13.2.43 to v17.1.1

CalvinWilkinson committed:
📦Update nuget package SixLabors.ImageSharp from version v2.0.0 to v2.1.3

CalvinWilkinson committed:
📦Update nuget package Silk.NET from version v2.6.0 to v2.16.0

CalvinWilkinson committed:
Merge pull request #337 from KinsonDigital/preview/feature/293-update-nuget-packages

Update Nuget Packages

CalvinWilkinson committed:
Start work for issue #344

CalvinWilkinson committed:
Update workflows to build and test with x64

CalvinWilkinson committed:
Update solution configs for x64 only

CalvinWilkinson committed:
Save run configs for JetBrains Rider users

CalvinWilkinson committed:
Project README updated

CalvinWilkinson committed:
Implement PR #344 Changes

CalvinWilkinson committed:
Create publish configurations for windows builds

CalvinWilkinson committed:
Create publish configurations for linux builds

CalvinWilkinson committed:
Implement PR #344 Changes

CalvinWilkinson committed:
Merge pull request #345 from KinsonDigital/preview/feature/344-setup-single-arch

Setup project to only run x64

CalvinWilkinson committed:
Start work for issue #342

CalvinWilkinson committed:
Remove unlinked solution item file reference

CalvinWilkinson committed:
Add a dotnet tool manifest to the project with the dotnet tool KinsonDigital.CICD tool

CalvinWilkinson committed:
Add NUKE parameters JSON file for default parameter setup

CalvinWilkinson committed:
Ignore local secrets JSON file that is used for local execution of CICD build system

CalvinWilkinson committed:
Create workflows for the CICD build system

quinnrdavis committed:
start work for issue #222

CalvinWilkinson committed:
Remove old workflow files

CalvinWilkinson committed:
Add words to dictionary for vscode

CalvinWilkinson committed:
Fix issue with OpenAL library not copying properly in GitHub CICD environment

quinnrdavis committed:
Removed mouse scroll speed from Test Mouse Scene

CalvinWilkinson committed:
Merge pull request #381 from quinnrdavis/preview/feature/222-remove-scroll-speed

Remove Mouse Scroll Speed

CalvinWilkinson committed:
Implement PR #379 Changes

CalvinWilkinson committed:
Merge pull request #379 from KinsonDigital/preview/feature/342-rework-build-system

🚧Change build system

CalvinWilkinson committed:
Start work for issue #382

CalvinWilkinson committed:
Refactor 'GetLastDirName' extension method to deal with pathing in a cross plat manner

CalvinWilkinson committed:
Refactor 'OnlyContainsDrive' extension method deal with pathing in a cross plat manner

CalvinWilkinson committed:
Refactor 'ContentPathResolver.RootDirectoryPath' prop and unit tests for cross plat paths

CalvinWilkinson committed:
Refactor 'ContentPathResolver.ResolveFilePath()' method with unit tests for cross plat paths

CalvinWilkinson committed:
Refactor 'ContentPathResolver.ResolveDirPath()' method with unit tests for cross plat paths

CalvinWilkinson committed:
Refactor the internal 'GetContentPath()' method for cross plat paths

CalvinWilkinson committed:
Simple cleanup of unit test resharper suppressions

CalvinWilkinson committed:
Refactor 'ContentPathResolver.ContentDirectoryName' test and add more tests

CalvinWilkinson committed:
Refactor 'TexturePathResolver' to deal with paths in cross plat manner

CalvinWilkinson committed:
Refactor 'SoundPathResolver' to deal with paths in a cross plat manner

CalvinWilkinson committed:
Refactor 'AtlasJSONDataPathResolver' to deal with paths in a cross plat manner

CalvinWilkinson committed:
Refactor 'ContentFontPathResolver' to deal with paths in a cross plat manner

CalvinWilkinson committed:
Refactor 'FontGlyphBatchItem.ToString()' tests to use cross plat new lines

CalvinWilkinson committed:
Refactor 'WindowsFontPathResolver' class and tests to deal with paths in a cross plat manner

CalvinWilkinson committed:
[cleanup] - Simple cleanup

CalvinWilkinson committed:
[cleanup] - Convert unit test class field to constant and rename for code policy

CalvinWilkinson committed:
Refactor 'AtlasLoader' class and tests to deal with paths in a cross plat manner

CalvinWilkinson committed:
Refactor 'ImageService' class and tests to deal with paths in a cross plat manner

CalvinWilkinson committed:
Replace all '\n' and '\r' new line characters with 'Environment.NewLine'

This is to keep things cross platform

CalvinWilkinson committed:
Refactor the unit test help class for loading test data to deal with paths in a cross plat manner

CalvinWilkinson committed:
Refactor the 'TextureLoaderTests' to use cross platform path dir separators

CalvinWilkinson committed:
[cleanup] - Do simple refactor and cleanup

CalvinWilkinson committed:
Refactor 'FontTests' unit tests to use cross plat pathing

CalvinWilkinson committed:
Refactor 'FontStatsTests' tests to use cross plat pathing

CalvinWilkinson committed:
Refactor 'SoundCacheTests' to use cross plat pathing

CalvinWilkinson committed:
Refactor 'TestHelpers' code to use cross plat pathing

CalvinWilkinson committed:
Refactor 'AtlasDataTests' to use cross plat pathing

CalvinWilkinson committed:
Refactor 'RenderTests' to use cross plat pathing

CalvinWilkinson committed:
Refactor 'TextureCacheTests' to use cross plat pathing

CalvinWilkinson committed:
Create new internal extension method to auto trim all forward and back slashes from a string

CalvinWilkinson committed:
Refactor 'AtlasData' to use cross plat pathing

CalvinWilkinson committed:
Refactor code to use 'TrimDirSeparatorFromEnd()' extension method

CalvinWilkinson committed:
Fix unit tests

CalvinWilkinson committed:
Improve 'GetLastDirName()' 'OnlyContainsDrive()' extension methods

CalvinWilkinson committed:
Disable use of IoC container when running tests

CalvinWilkinson committed:
Create theories for running tests that only run on a particular OS

CalvinWilkinson committed:
Refactor 'GetLastDirName' extension method for cross plat pathing

CalvinWilkinson committed:
Fix unit test

CalvinWilkinson committed:
Implement PR #383 Changes

CalvinWilkinson committed:
Merge pull request #383 from KinsonDigital/preview/feature/382-fix-pathing-issues

🚧Fix pathing issues

CalvinWilkinson committed:
Start work for issue #385

CalvinWilkinson committed:
Create extension method with tests to help normalize paths

CalvinWilkinson committed:
[bug] - Fix pathing bug in sound path resolver

CalvinWilkinson committed:
[bug] - Fix pathing bug in texture path resolver

CalvinWilkinson committed:
[bug] - Fix pathing bug in content font path resolver

CalvinWilkinson committed:
[bug] - Fix pathing bug in atlas JSON data path resolver

CalvinWilkinson committed:
[bug] - Fix pathing bug in windows font path resolver

Also throw exception if being used on a non windows operating system

CalvinWilkinson committed:
Normalize ctor code docs for 'IPlatform' parameter

CalvinWilkinson committed:
Merge pull request #386 from KinsonDigital/preview/feature/385-fix-path-resolvers

🚧Fix bug with path resolvers

CalvinWilkinson committed:
Update styling for v1.0.0-preview.3.md release notes

CalvinWilkinson committed:
Add words to dictionary for VSCode

CalvinWilkinson committed:
Create release notes for v1.0.0-preview.10

kselena commented on Documentation/ReleaseNotes/PreviewReleases/Release-Notes-v1.0.0-preview.10.md:
```diff
@@ -0,0 +1,85 @@
+<h2 style="font-weight:bold" align="center">Other 🪧</h2>
+<h5 align="center">(Includes anything that does not fit into the categories above)</h5>
+
+1. [#310](https://github.com/KinsonDigital/Velaptor/issues/310) - Made simple grammar change to the project's readme document.
```
grammar
```suggestion
1. [#310](https://github.com/KinsonDigital/Velaptor/issues/310) - Made simple grammar changes to the project's readme document.
```

CalvinWilkinson committed:
Remove old and unused vscode snippets file

CalvinWilkinson committed:
Add vscode snippets file to assist with building release notes

CalvinWilkinson committed:
Add missing issue to the v1.0.0-preview.10 release notes

CalvinWilkinson committed:
Update the kinsondigital.cicd dotnet tool from version v1.0.0-preview.11 to v1.0.0-preview.12

This was to bring in a bug fix and was preventing Velaptor from having a correct status check run

CalvinWilkinson committed:
Fix URL problem with release notes issue link

CalvinWilkinson committed:
Fix URL's for issue snippets
